src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureServiceBus/AzureServiceBusReceiveWrapper.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.Reflection;
+
+namespace NewRelic.Providers.Wrapper.AzureServiceBus;
+
+public class AzureServiceBusReceiveWrapper : AzureServiceBusWrapperBase
+{
+    private static readonly ConcurrentDictionary<Type, Func<object, object>> _getResultFromGenericTask = new();
+
+    public override CanWrapResponse CanWrap(InstrumentedMethodInfo instrumentedMethodInfo)
+    {
+        var canWrap = instrumentedMethodInfo.RequestedWrapperName.Equals(nameof(AzureServiceBusReceiveWrapper));
+        return new CanWrapResponse(canWrap);
+    }
+
+    public override AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall, IAgent agent, ITransaction transaction)
+    {
+        dynamic serviceBusReceiver = instrumentedMethodCall.MethodCall.InvocationTarget;
+        string queueName = serviceBusReceiver.EntityPath; // marty-test-queue
```

Comment:
Like the examples, but should these be generic and not specific to your testing?

Reply:
Done